tests/test_estimation.py

```diff
@@ -1,91 +1,206 @@
-                                    baseline_cov=None)
+    n_classes, n_trials, n_channels, n_times = 2, 4, 6, 100
+    x = rndstate.randn(n_trials, n_channels, n_times)
+    labels = np.array([0, 1]).repeat(n_trials // n_classes)
+    cov = XdawnCovariances(nfilter=nfilter)
+    covmats = cov.fit_transform(x, labels)
+    assert cov.get_params() == dict(
+        nfilter=nfilter,
+        applyfilters=True,
+        classes=None,
+        estimator="scm",
+        xdawn_estimator="scm",
+        baseline_cov=None,
+    )
+    covsize = 2 * (n_classes * nfilter)
+    assert covmats.shape == (n_trials, covsize, covsize)
+    assert np.any(np.linalg.eigvals(covmats) > 0.0)
 
 
-def test_cosp_covariances():
+def test_xdawn_covariances_applyfilters(rndstate):
+    n_classes, nfilter = 2, 2
+    n_trials, n_channels, n_times = 4, 6, 100
+    x = rndstate.randn(n_trials, n_channels, n_times)
+    labels = np.array([0, 1]).repeat(n_trials // n_classes)
+    cov = XdawnCovariances(nfilter=nfilter, applyfilters=False)
+    covmats = cov.fit_transform(x, labels)
+    covsize = n_classes * nfilter + n_channels
+    assert covmats.shape == (n_trials, covsize, covsize)
+    assert np.any(np.linalg.eigvals(covmats) > 0.0)
+
+
+def test_cosp_covariances(rndstate):
     """Test fit CospCovariances"""
-    x = np.random.randn(2, 3, 1000)
+    n_trials, n_channels, n_times = 2, 3, 1000
+    x = rndstate.randn(n_trials, n_channels, n_times)
     cov = CospCovariances()
     cov.fit(x)
-    cov.fit_transform(x)
-    assert cov.get_params() == dict(window=128, overlap=0.75, fmin=None,
-                                    fmax=None, fs=None)
+    covmats = cov.transform(x)
+    assert cov.get_params() == dict(
+        window=128, overlap=0.75, fmin=None, fmax=None, fs=None
+    )
+    n_freqs = 65
+    assert covmats.shape == (n_trials, n_channels, n_channels, n_freqs)
+    assert np.any(np.linalg.eigvals(covmats.mean(axis=-1)) > 0.0)
 
-@pytest.mark.parametrize('coh',
-    ['ordinary', 'instantaneous', 'lagged', 'imaginary']
-)
-def test_coherences(coh):
+
+@pytest.mark.parametrize("coh", coh)
+def test_coherences(coh, rndstate):
     """Test fit Coherences"""
-    rs = np.random.RandomState(42)
-    n_trials, n_channels, n_times = 10, 3, 1000
-    x = rs.randn(n_trials, n_channels, n_times)
+    n_trials, n_channels, n_times = 2, 3, 1000
+    x = rndstate.randn(n_trials, n_channels, n_times)
 
     cov = Coherences(coh=coh)
     cov.fit(x)
-    cov.fit_transform(x)
-    assert cov.get_params() == dict(window=128, overlap=0.75, fmin=None,
-                                    fmax=None, fs=None, coh=coh)
+    covmats = cov.fit_transform(x)
+    assert cov.get_params() == dict(
+        window=128, overlap=0.75, fmin=None, fmax=None, fs=None, coh=coh
+    )
+    n_freqs = 65
+    assert covmats.shape == (n_trials, n_channels, n_channels, n_freqs)
+    assert np.any(np.linalg.eigvals(covmats.mean(axis=-1)) > 0.0)
```

Comment:
Why do you apply an average on the last axis of `covmats`?

Reply:
I don't know if instantaneous, lagged and imaginary coherence matrices are SPSD.

Reply:
yes, they are ;) This will be the subject of another PR